lib/src/sync/warp_sync.rs

```diff
@@ -620,9 +659,24 @@ impl<TSrc, TRq> InProgressWarpSync<TSrc, TRq> {
         let runtime_parameters_get = if let Phase::RuntimeDownload {
             header,
             warp_sync_source_id,
+            hint_doesnt_match,
+            downloaded_runtime: None,
```

Comment:
This specific line fixes a bug mentioned in the CHANGELOG where the warp syncing downloads the runtime code twice.